### PR TITLE
DIV-4945 / DIV-5490 Solicitor role update / Case event auth validation

### PR DESCRIPTION
**Pull request by alexwakeman**
- Ensure that new solicitor role is added to DN submit event
- Ensure removal of old callback (not needed)
- Add unit testing for case event validation

[Create validation test on CCD config](https://tools.hmcts.net/jira/browse/DIV-5490)

[Solicitor can only see events in CCD relative to whom they represent](https://tools.hmcts.net/jira/browse/DIV-4945)

### CODE REVIEW

File: test/unit/definitions/divorce/UserEventAuthorisation.test.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  });
+
+  it('should have at least CRU or RU access level for all MANDATORY, OPTIONAL and READONLY show/hide event fields', () => {
+    caseEventsActive.forEach(eventAuth => {
+      const userRole = eventAuth.UserRole;
+      const eventName = eventAuth.CaseEventID;
+      const caseType = eventAuth.CaseTypeID;
+
+      // get all MANDATORY and OPTIONAL fields for this event
+      let caseFieldsForEvent = caseEventFields.filter(entry => {
```
Comment (yarpo) [inline test/unit/definitions/divorce/UserEventAuthorisation.test.js]: can you please extract logic to named function and make sure if this name tells what function doeas

Comment (alexwakeman) [reply test/unit/definitions/divorce/UserEventAuthorisation.test.js]: Done 👍

File: package.json [hunk 2]
```diff
@@ -33,7 +33,7 @@
     }
   },
   "scripts": {
-    "setup": "git submodule update --init --recursive && git submodule foreach --recursive git pull origin master && yarn --cwd ccd-definition-processor install",
+    "setup": "git submodule update --init --recursive && git submodule foreach git reset --hard 4d70e97 && yarn --cwd ccd-definition-processor install",
```
Comment (yarpo) [inline package.json]: are you sure this is what we want?

Comment (alexwakeman) [reply package.json]: Hi @yarpo yes we need this to allow the pipeline to pass. The newest version of the submodule has a breaking change, so this fixes the version to one that works for us. This submodule is not ours so we can't control versions and breaking changes.

File: definitions/divorce/json/CaseEvent.json [hunk 3]
```diff
@@ -972,9 +972,7 @@
     "CallBackURLAboutToSubmitEvent": "${CCD_DEF_COS_URL}/solicitor-update",
     "RetriesTimeoutURLAboutToSubmitEvent": "120,120",
     "SecurityClassification": "Public",
-    "ShowSummary": "Y",
-    "CallBackURLAboutToSubmitEvent": "${CCD_DEF_COS_URL}/solicitor-update",
-    "RetriesTimeoutURLAboutToSubmitEvent": "120,120"
```
Comment (hosainnet) [inline definitions/divorce/json/CaseEvent.json]: I guess this was a duplicate?

File: definitions/divorce/json/CaseEventToFields.json [hunk 4]
```diff
@@ -4656,14 +4656,6 @@
...
     "PageID": 1
   },
-  {
-    "LiveFrom": "01/01/2017",
-    "CaseTypeID": "DIVORCE",
-    "CaseEventID": "create",
-    "CaseFieldID": "AosLetterHolderId",
-    "DisplayContext": "OPTIONAL",
-    "PageID": 1
-  },
```
Comment (hosainnet) [inline definitions/divorce/json/CaseEventToFields.json]: This is the unused one right?

File: definitions/divorce/json/CaseEventToFields.json [hunk 5]
```diff
@@ -7083,7 +7075,7 @@
     "LiveFrom": "16/04/2019",
     "CaseTypeID": "DIVORCE",
     "CaseEventID": "solicitorSOTDN",
-    "CaseFieldID": "SeparationTimeLivedTogetherDetailsDN\t",
```
Comment (hosainnet) [inline definitions/divorce/json/CaseEventToFields.json]: WTH did this ever work

File: package.json (hunk 2 above)
Comment (hosainnet) [reply package.json]: I'm not sure we should be fixing it, if we just remove `git submodule foreach --recursive git pull origin master` it'll use whatever reference is in the `ccd-definition-processor` file which already has a git reference

File: definitions/divorce/json/CaseEvent.json (hunk 3 above)
Comment (alexwakeman) [reply definitions/divorce/json/CaseEvent.json]: yep sorry I should probably leave more comments

File: test/unit/definitions/divorce/GenericValidation.test.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  FixedLists: Object.assign(require('definitions/divorce/json/FixedLists'), []),
+  Jurisdiction: Object.assign(require('definitions/divorce/json/Jurisdiction'), []),
+  SearchAlias: Object.assign(require('definitions/divorce/json/SearchAlias'), []),
+  SearchInputFields: Object.assign(require('definitions/divorce/json/SearchInputFields'), []),
+  SearchResultFields: Object.assign(require('definitions/divorce/json/SearchResultFields'), []),
+  State: Object.assign(require('definitions/divorce/json/State'), []),
+  UserProfile: Object.assign(require('definitions/divorce/json/UserProfile'), []),
+  WorkBasketInputFields: Object.assign(require('definitions/divorce/json/WorkBasketInputFields'), []),
+  WorkBasketResultFields: Object.assign(require('definitions/divorce/json/WorkBasketResultFields'), [])
+};
```
Comment (hosainnet) [inline test/unit/definitions/divorce/GenericValidation.test.js]: The mother of all tests lol

File: definitions/divorce/json/CaseEventToFields.json (hunk 4 above)
Comment (alexwakeman) [reply definitions/divorce/json/CaseEventToFields.json]: yes its unused for this event

File: definitions/divorce/json/CaseEventToFields.json (hunk 5 above)
Comment (alexwakeman) [reply definitions/divorce/json/CaseEventToFields.json]: lol yeah I think it's only used on the front-end in these instances, other references used elsewhere don't have the tab

File: package.json (hunk 2 above)
Comment (alexwakeman) [reply package.json]: cool - has been reverted now and we have the original sub-module commit in this branch

File: test/unit/definitions/divorce/GenericValidation.test.js (hunk 6 above)
Comment (yarpo) [reply test/unit/definitions/divorce/GenericValidation.test.js]: One test to validate them all 💍

**Pull request closed without merging**